client/trino-client/src/test/java/io/trino/client/auth/external/TestExternalAuthenticator.java

```diff
@@ -101,16 +105,56 @@ public void testChallengeWithMalformedUri()
                 .hasRootCauseMessage("Malformed IPv6 address at index 8: http://[1.1.1.1]");
     }
 
+    @Test
+    public void testAuthentication()
+    {
+        MockTokenPoller tokenPoller = new MockTokenPoller()
+                .withResult(URI.create("http://token.uri"), successful(new Token("valid-token")));
+        ExternalAuthenticator authenticator = new ExternalAuthenticator(uri -> {}, tokenPoller, Duration.ofSeconds(1));
+
+        Request authenticated = authenticator.authenticate(null, getUnauthorizedResponse("Bearer x_token_server=\"http://token.uri\""));
+
+        assertThat(authenticated.headers())
+                .extracting(headers -> headers.get(AUTHORIZATION))
+                .isEqualTo("Bearer valid-token");
+    }
+
+    @Test
+    public void testReAuthenticationAfterRejectingToken()
```

Comment:
nit: You could add this tests in a commit before, exposing the issue. Then with fix you could show how the behavior changed here.